### PR TITLE
added function to routesfavorite

### PR DESCRIPTION
**Pull request by go-to-stars**

### CODE REVIEW

**Review by Stylize-K — APPROVED** [review]

File: controllers/drinks.js [hunk 1]
```diff
@@ -264,7 +265,7 @@ const addDrink = async (req, res, next) => {
   );
 
   if (!response) {
-    throw HttpError(404, "Add request not made (invalid request)");
+    next(HttpError(404, "Add request not made (invalid request)"));
   }
 
   res.status(200).json({
```
Comment (Illya-Bilyk) [inline controllers/drinks.js:271]: I don't see the function.

File: controllers/drinks.js [hunk 2]
```diff
@@ -298,25 +299,134 @@ const getDrink = async (req, res) => {
...
+  // }
+
+  const responseUser = await User.updateOne(
+    { _id: userId },
+    { $addToSet: { favoritesDrink: drinkId } }
+  );
+
+  if (!responseUser) {
+    next(HttpError(404, "Add request not made (invalid request)"));
+  }
```
Comment (Illya-Bilyk) [inline controllers/drinks.js:322]: Better to say just "Not found"

File: controllers/drinks.js [hunk 3]
```diff
@@ -298,25 +299,134 @@ const getDrink = async (req, res) => {
...
+
+  const userFavoritesDrink = await User.findById(userId);
+
+  if (!userFavoritesDrink) {
+    next(HttpError(404, "Add request not made (invalid request)"));
```
Comment (Illya-Bilyk) [inline controllers/drinks.js:327]: Better to say just "Not found"

File: controllers/drinks.js [hunk 4]
```diff
@@ -298,25 +299,134 @@ const getDrink = async (req, res) => {
...
+  // }
+
+  const responseUser = await User.updateOne(
+    { _id: userId },
+    { $pull: { favoritesDrink: drinkId } }
+  );
+
+  if (!responseUser) {
+    next(HttpError(404, "Add request not made (invalid request)"));
+  }
```
Comment (Illya-Bilyk) [inline controllers/drinks.js:355]: Better to say just "Not found"

File: controllers/drinks.js [hunk 5]
```diff
@@ -298,25 +299,134 @@ const getDrink = async (req, res) => {
...
+
+  const userFavoritesDrink = await User.findById(userId);
+
+  if (!userFavoritesDrink) {
+    next(HttpError(404, "Add request not made (invalid request)"));
+  }
```
Comment (Illya-Bilyk) [inline controllers/drinks.js:361]: Better to say just "Not found"

File: controllers/drinks.js [hunk 6]
```diff
@@ -298,25 +299,134 @@ const getDrink = async (req, res) => {
...
+    "-createdAt -updatedAt",
+    {
+      skip,
+      limit,
+    }
+  );
+
+  if (!response) {
+    next(HttpError(404, "Search query not found (invalid query)"));
+  }
```
Comment (Illya-Bilyk) [inline controllers/drinks.js:413]: Better to say just "Not found"

File: controllers/drinks.js [hunk 7]
```diff
@@ -298,25 +299,134 @@ const getDrink = async (req, res) => {
...
-const getDrinkById = async (req, res) => {
+const getDrinkById = async (req, res, next) => {
   const { drinkId } = req.params;
 
   const response = await Recipe.findById(drinkId);
 
   if (!response) {
-    throw HttpError(404, "Drink not found");
+    next(HttpError(404, "Drink not found"));
   }
```
Comment (Illya-Bilyk) [inline controllers/drinks.js:430]: Better to say just "Not found"

**Review by Illya-Bilyk — APPROVED** [review]

**Pull request closed without merging**